src/python/WMCore/Services/Requests.py

```diff
@@ -255,6 +255,13 @@ def makeRequest_httplib(self, uri=None, data={}, verb='GET',
                 # socket/httplib really screwed up - nuclear option
                 conn.connections = {}
                 raise socket.error('Error contacting: %s: %s' % (self.getDomainName(), msg))
+        except Exception as ex:
+            if "Unable to find the server at" in str(ex):
```

Comment:
I see. I think it would be better to check the exception unless pycurl returns the same string. It would be clearer to change when we move to pycurl.
As far as switching to HTTP 400 error, I am not sure where it is the case for server is down or client is sending wrong address, I guess it would be mostly first, then we shouldn't convert to 400 error.